### PR TITLE
fixed nwb

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
1. Fixed the NWB data type issue for `B_DelayStartTimeHarp`. The `B_DelayStartTimeHarp` is `Null` for optogenetics (no delay period for optogenetics condition), which will cause an error. This issue is solved by setting the default to `np.nan` for empty `B_DelayStartTimeHarp`.
2. Handle different names. `Laser_#` was previously updated to `LaserColor_#` for clarity. Updated here to handle both names. 

### What issues or discussions does this update address?
see Describe changes section. 

### Describe the expected change in behavior from the perspective of the experimenter
No

### Describe any manual update steps for task computers
No

### Was this update tested in 446/447?
Tested on my own computer with multiple sessions.

### CODE REVIEW

File: src/foraging_gui/TransferToNWB.py
Comment (alexpiet) [inline src/foraging_gui/TransferToNWB.py:1]: Is this file being used anymore, once the automatic pipeline starts running? @hanhou

Comment (hanhou) [reply src/foraging_gui/TransferToNWB.py:1]: My temporary pipeline is using this file, while the automatiic pipeline under development is more complicated. They copy and paste our code and develop separately in `/util/behavior_to_nwb.py` file in [this capsule](https://codeocean.allenneuraldynamics.org/capsule/9216710/tree) of [this pipeline](https://codeocean.allenneuraldynamics.org/capsule/0995759/tree). We should definitely find a better place to put this file. @rachelstephlee  @Ahad-Allen

**Comment by XX-Yin** [comment]
@alexpiet @hanhou I think we can merge this PR first and then migrate it to a different place where the downstream pipeline can easily get access to.

**Review by hanhou — APPROVED** [review]

**Review by hanhou — COMMENTED** [review]
Please also add a new json test file [here](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/377) that this PR was supposed to fix, and include it to the test list at line 577.

**Comment by XX-Yin** [comment]
Added, but that file is not added to the `test_json_urls` in the `TransferToNWB.py`. We can do it in the next update.

**Pull request merged**